### PR TITLE
util: add instructions for running the VM on macOS

### PR DESCRIPTION
**Pull request by razvanvirtan**

### CODE REVIEW

File: util/macos-vm/README.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+Make sure `qemu` is installed on your system:
+
+```
+brew install qemu
+```
+
+## Running the VM
+
+First, download the `so-ubuntu-20-04.img` from [here](https://repository.grid.pub.ro/cs/so/linux-2024/).
```
Comment (fabianpatras) [inline util/macos-vm/README.md:13]:
```suggestion
First, download `so-ubuntu-20-04.img` from [here](https://repository.grid.pub.ro/cs/so/linux-2024/).
```

Comment (fabianpatras) [reply util/macos-vm/README.md:13]: Can you also give a warning about the file size?
I can see it is quite big, at ~8.8GB. Thanks.

File: util/macos-vm/README.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+Then, use the `qemu_run.sh` script to run the VM:
+```
+sudo ./qemu_run.sh <path_to_vm_image> <number_of_cores> <number_of_GB_for_the_VM_RAM> <host_interface>
+```
+
+The `path_to_vm_image` parameter is the path where you downloaded the `so-ubuntu-20-04.img` file.
```
Comment (fabianpatras) [inline util/macos-vm/README.md:20]: Can we replace `The <bla bla paramter>` with a bullet point list?
This way we would avoid repeating `The` word and the resulting markdown IMO will look better.

File: util/macos-vm/README.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+The `number_of_cores` and `number_of_GB_for_the_VM_RAM` parameters are self-explanatory.
```
Comment (fabianpatras) [inline util/macos-vm/README.md:22]: Can we refrain from using phrases like `self-explanatory` in a step-by-step tutorial?

One possible rephrasing I can think about is to rename both parameters into simpler words and then explain what they mean.
```suggestion
- `cores` represents the number of cores you want the VM to use. <possible default or recommendations?>
- `memory` represents how many GB of RAM you want the VM to use. <possible default or recommendations?>
```

Thanks

**Review by fabianpatras — COMMENTED** [review]
I left a few comments regarding the overall phrasing.

**Comment by razvanvirtan** [comment]
It looks like `superlinter.sh` and `spellchecking` don't agree on the correct spelling for WiFi. `superlinter.sh` asks for `Wi-Fi`, while `spellchecking` asks for `WiFi`. That is the reason for the remaining CI error.

File: util/macos-vm/README.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+Hardware Port: Thunderbolt Bridge
+Device: bridge0
+Ethernet Address: 36:a1:9b:9f:e8:80
+
+Hardware Port: Wi-Fi
+Device: en0
+Ethernet Address: bc:d0:74:ad:8a:b1
+```
+
+Therefore, we choose `en0` as the value of the `host_interface` parameter.
```
Comment (Alex-deVis) [inline util/macos-vm/README.md:52]:
````suggestion
Therefore, we choose `en0` as the value of the `host_interface` parameter.

Using the above recommendations, the command to start the VM will look like this:

```console
sudo ./qemu_run.sh so-ubuntu-20-04.img 2 2 en0
```
````

File: util/macos-vm/README.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+- `host_interface`: the host interface used to provide the VM with internet access
+
+In order to find the value for `host_interface`, you can use the following command on your macOS host.
+
+```console
+networksetup -listallhardwareports
+```
+
+The value of `host_interface` will be the device name of one of your WiFi or ethernet ports, according to your type of connection.
+As an example, for a macOS host connected to WiFi, we could get the following `networksetup` output:
```
Comment (Alex-deVis) [inline util/macos-vm/README.md:36]:
```suggestion
The value of `host_interface` will be the device name of one of your Wi-Fi or ethernet ports, according to your type of connection.
As an example, for a macOS host connected to Wi-Fi, we could get the following `networksetup` output:
```
Fix this linter issue.

**Review by Alex-deVis — APPROVED** [review]
Address the inline comments and merge it afterward, we need this ASAP.

**Comment by razvanvirtan** [comment]
@Alex-deVis The inline comments are solved. However, solving the linter errors lead to spellcheck errors, as described [here](https://github.com/cs-pub-ro/operating-systems/pull/10#issuecomment-1773761866). If this is ok, we can merge, otherwise I think we should solve the problem in the CI config.

**Pull request merged**